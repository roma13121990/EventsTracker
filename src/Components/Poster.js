import React from 'react';
import FaFacebookSquare from 'react-icons/lib/fa/facebook-square'
export default class Poster extends React.Component {
    render() {
        let {name, facebook_page_url, thumb_url} = this.props.artist;
        return (
            <div className="Poster col-sm-6 col-md-3 col-lg-3 col-xs-12">
                <div>
                    <section><img src={thumb_url} alt={name} width="200" height="200" className="mx-auto d-block"/></section>
                    <aside><p>{name}</p>
                        {facebook_page_url !== "" && <a href={facebook_page_url} target="_blank"><FaFacebookSquare style={{color: '#4266b2',fontSize: 25}}/></a>}
                    </aside>
                </div>
            </div>
        )
    }
}