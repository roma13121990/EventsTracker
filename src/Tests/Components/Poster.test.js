import React from 'react'
import renderer from 'react-test-renderer'
import Poster from "../../Components/Poster";

describe('<Poster />', () => {
    const propsWithFB = {
        "artist":
            {
                "thumb_url": "https://s3.amazonaws.com/bit-photos/thumb/215115.jpeg",
                "mbid": "03746227-c9f9-4b3b-b2c6-9b31f79564f7",
                "facebook_page_url": "",
                "image_url": "https://s3.amazonaws.com/bit-photos/large/215115.jpeg",
                "name": "Selena", "id": "26427",
                "tracker_count": 303472, "upcoming_event_count": 0,
                "url": "https://www.bandsintown.com/a/26427?came_from=267&app_id=ID"
            }
    };

    const propsWithoutFB = {
        "artist":
            {
                "thumb_url":"https://s3.amazonaws.com/bit-photos/thumb/8458441.jpeg","mbid":"",
                "facebook_page_url":"https://www.facebook.com/dualipaofficial/",
                "image_url":"https://s3.amazonaws.com/bit-photos/large/8458441.jpeg",
                "name":"Dua Lipa","id":"1750816","tracker_count":196950,"upcoming_event_count":74,
                "url":"https://www.bandsintown.com/a/1750816?came_from=267&app_id=ID"
            }
    };

    it('should generate snapshot for Poster component with FB Url', () => {
        const tree = renderer.create(<Poster {...propsWithFB}/>).toJSON();
        expect(tree).toMatchSnapshot()
    })

    it('should generate snapshot for Poster component without FB url', () => {
        const tree = renderer.create(<Poster {...propsWithoutFB}/>).toJSON();
        expect(tree).toMatchSnapshot()
    })
});