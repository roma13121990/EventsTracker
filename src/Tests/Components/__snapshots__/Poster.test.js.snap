// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Poster /> should generate snapshot for Poster component with FB Url 1`] = `
<div
  className="Poster col-sm-6 col-md-3 col-lg-3 col-xs-12"
>
  <div>
    <section>
      <img
        alt="Selena"
        className="mx-auto d-block"
        height="200"
        src="https://s3.amazonaws.com/bit-photos/thumb/215115.jpeg"
        width="200"
      />
    </section>
    <aside>
      <p>
        Selena
      </p>
    </aside>
  </div>
</div>
`;

exports[`<Poster /> should generate snapshot for Poster component without FB url 1`] = `
<div
  className="Poster col-sm-6 col-md-3 col-lg-3 col-xs-12"
>
  <div>
    <section>
      <img
        alt="Dua Lipa"
        className="mx-auto d-block"
        height="200"
        src="https://s3.amazonaws.com/bit-photos/thumb/8458441.jpeg"
        width="200"
      />
    </section>
    <aside>
      <p>
        Dua Lipa
      </p>
      <a
        href="https://www.facebook.com/dualipaofficial/"
        target="_blank"
      >
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": "#4266b2",
              "fontSize": 25,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em"
        >
          <g>
            <path
              d="m30.9 2.9q2.6 0 4.5 1.8t1.9 4.6v21.4q0 2.7-1.9 4.6t-4.5 1.8h-4.2v-13.2h4.4l0.7-5.2h-5.1v-3.3q0-1.3 0.5-1.9t2-0.6l2.8 0v-4.7q-1.5-0.2-4-0.2-3.1 0-4.9 1.8t-1.8 5.1v3.8h-4.5v5.2h4.5v13.2h-11.9q-2.6 0-4.5-1.8t-1.9-4.6v-21.4q0-2.7 1.9-4.6t4.5-1.8h21.5z"
            />
          </g>
        </svg>
      </a>
    </aside>
  </div>
</div>
`;
